### ISSUE TITLE
Custom prime testing implementation

### ISSUE THREAD

**Issue by battila7** [cryptid-org/cryptid-native#24]
## Background

[GMP-LAWL](https://github.com/cryptid-org/gmp-lawl/) is intended to be a GMP-based minimal maths lib. Thus, randomgen is completely out of scope for that project. However, CryptID.native currently makes use of the `mpz_probab_prime_p` function which in turn uses an implementation of the Miller-Rabin primality test involving GMP-based random routines.

As randomgen cannot be a part of GMP-LAWL, this situation must be resolved by changes on the CryptID.native side.

## Task

Implement a primality test which makes use of our own randomgen routines. The actual code should be based on the GMP implementation.

**Comment by battila7** [comment]
Note, that the development of GMP-LAWL can be blocked by this issue if not resolved in the near future.

**Comment by BeardOfDoom** [comment]
After a brief discussion we decided to only implement a miller-rabin primality test without trial division before it. The reason is that, with this choice we can keep our function calls in the "mpz" layer, so the work on gmp-lawl might be easier. 

Don't close this issue after the miller-rabin is implemented, to remind us it is not a perfect solution yet.

**Comment by battila7** [comment]
Once LAWL is integrated (#16), we should revisit this issue.
